internal/handlers/common/aggregations/stages/project.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stages
+
+import (
+	"context"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/types"
+)
+
+// project represents $project stage.
+//
+//	{
+//	  $project: {
+//	    <output field1>: <expression1>,
+//	    ...
+//	    <output fieldN>: <expressionN>
+//	  }
+type project struct {
+	projection *types.Document
+	inclusion  bool
+}
+
+// newProject validates projection document and creates a new $project stage.
+func newProject(stage *types.Document) (Stage, error) {
+	fields, err := common.GetRequiredParam[*types.Document](stage, "$project")
+	if err != nil {
+		return nil, err
```

Comment:
Could you add a test for `$project` not being document? Something like below would give us error message here.

```
pipeline: bson.A{bson.D{{"$project", "invalid"}}}
```

Reply:
Nice catch, thanks! Added this one.